data/balance.go

```diff
@@ -53,22 +54,27 @@ func (s BalanceSlice) Less(i, j int) bool {
 	}
 }
 
-func (s *BalanceSlice) Add(counterparty *Account, balance, change *Value, currency *Currency) {
-	*s = append(*s, Balance{*counterparty, *balance, *change, *currency})
+func (s *BalanceSlice) Add(counterparty *Account, balance, change *Value, currency *Currency, AMMID *Hash256) {
```

Comment:
This is an API change that would break existing code.

Reply:
Aware of that. The underlying problem is the design of the XRPL in this place: They just added this AMMID at a rather random location (observing from the outisde). I do not have a better solution unless we add "if amm then" in multiple places.